README.rst

```diff
@@ -258,6 +258,34 @@ For accounts in regions outside of US-West, use :code:`region` parameter to spec
     $dbh = new PDO("snowflake:account=testaccount.us-east-1", "user", "password");
     $dbh = new PDO("snowflake:account=testaccount;region=us-east-1", "user", "password");
 
+Using Key Pair Authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The PHP PDO driver supports key pair authentication and key rotation.
+
+You must first complete the initial configuration for key pair authentication as shown in 
+`Key Pair Authentication & Key Pair Rotation <https://docs.snowflake.com/en/user-guide/key-pair-auth.html#key-pair-authentication-key-pair-rotation>`_.
+
+To connect to the Snowflake database using key par authentication, create a new :code:`PDO` object, as explained in the
+`PHP PDO documentation <https://www.php.net/manual/en/pdo.connections.php>`_.
+Specify the data source name (:code:`dsn`) parameter as shown below:
+
+.. code-block:: php
+
+    $dbh = new PDO("snowflake:account=<account_name>", 
+                   "AUTHENTICATOR = SNOWFLAKE_JWT", 
+                   "PRIV_KEY_FILE = <path>/rsa_key.p8`",
+                   "PRIV_KEY_FILE_PWD = <private_key_passphrase>");
```

Comment:
I think this syntax is not correct.

1. This example shows each DSN property (AUTHENTICATOR, PRIV_KEY_FILE, PRIV_KEY_FILE_PWD) being passed as a separate argument.
2. The example doesn't show the username being passed in as a separate argument.

I think that:

1. Each of the new DSN properties needs to be part of the first argument and delimited by semicolons.  The [test case](https://github.com/snowflakedb/pdo_snowflake/pull/289/files#diff-0272c80a6ed5c366c86659fb7177d2788789fd918781941e3e55f9b57380f928) seems to indicate that this is how you should specify this.
2. In the test case also shows the username being passed in as a separate argument. I think in all of our drivers, when we authenticate using a private key, we still need to specify a username (e.g. the [Python Connector](https://docs.snowflake.com/en/user-guide/python-connector-example.html#using-key-pair-authentication-key-pair-rotation) example).

    The test cases also pass in the password argument, which makes me wonder if the argument is required and if it should be an empty string if you're using a private key (rather than a password) for authentication.

Reply:
Dev approved it, but I will run it by them again

Reply:
I think we're providing our own implementation of this interface, and the constructor doesn't let you pass in arguments specifically for the authenticator, private key file, and private key file passphrase:

https://www.php.net/manual/en/pdo.connections.php
https://www.php.net/manual/en/pdo.construct.php

Any other information that needs to be passed in needs to go into the argument for the DSN or possibly the options array.

Reply:
Reaching out to dev again.

Reply:
I think the example is still wrong. The example passes in five arguments:

- The string containing the DSN.
- The string containing the username (but in this odd "username=<username>" format, which doesn't seem to be correct).
- The string containing the authenticator parameter.
- The string containing the private key file parameter.
- The string containing the private key passphrase parameter.

We point to the PDO documentation, but their docs don't have anything about a constructor that passes in five string arguments:

https://www.php.net/manual/en/pdo.connections.php
https://www.php.net/manual/en/pdo.construct.php

I think the example is supposed to look like this:

```
$dbh = new PDO("snowflake:account=<account_name>; authenticator=snowflake_jwt;priv_key_file=<path_to_private_key_file.;priv_key_file_pwd=<private_key_passphrase>", "<user>", "<password>");
```

If the "<password>" argument is required, I'm assuming that it should be a blank string if you're using the key pair for authentication.

Reply:
I got a new example from dev (he apologized for not checking it closely) and am integrating it.